File: buff/classes/354/pub/etc/student.gccgo/student.go
package person

type IStudent interface {
	IPerson						// embedded interface: superclass
}

type Student struct {
	Person						// embedded struct: superclass
	major string
}

func NewStudent(name string, age int, major string) (s *Student) {
	s=new(Student)
	s.Init(name,age,major)
	return
}

func (s *Student) Init(name string, age int, major string) {
	s.Person.Init(name,age)		// superclass constructor
	s.major=major
}

func (s *Student) String() string {
	return s.Person.String()+" "+s.major // super.String()
}

File: buff/classes/354/pub/etc/Student/Student.java
public class Student extends Person {

    private String major;

    public Student(String name, int age, String major) {
        super(name,age);
        init(major);
    }

    private void init(String major) {
        this.major=major;
    }

    protected String string() { return super.string()+" "+major; }

    public String toString() { return string(); }

}
